src/net_processing.cpp

```diff
@@ -2866,7 +2870,7 @@ void PeerManagerImpl::ProcessHeadersMessage(CNode& pfrom, Peer& peer,
         // Headers message had its maximum size; the peer may have more headers.
         if (MaybeSendGetHeaders(pfrom, GetLocator(pindexLast), peer)) {
             LogPrint(BCLog::NET, "more getheaders (%d) to end to peer=%d (startheight:%d)\n",
-                    pindexLast->nHeight, pfrom.GetId(), peer.m_starting_height);
+                     pindexLast->nHeight, pfrom.GetId(), peer.m_starting_height.value());
```

Comment:
If `m_starting_height` has no value the call to `value()` will throw an exception. I do not feel comfortable either assuming that it will always have value here (also considering future changes) or that throwing an exception is ok. Some of this code is surrounded by `try/catch` higher up, but we may move pieces of code around, possibly moving this printout elsewhere. Maybe do it more defensively:

```cpp
LogPrint(BCLog::NET, "more getheaders (%d) to end to peer=%d%s\n",
         pindexLast->nHeight,
         pfrom.GetId(),
         peer.m_starting_height.has_value() ? strprintf(" (startheight:%d)", peer.m_starting_height.value()) : "");
```

(here and in the other printouts)